config/config.go

```diff
@@ -195,19 +196,86 @@ func ensureRepo(repo, dir string) (*vcs.GitRepo, error) {
 // UpdateAll does a git fast-forward pull from each remote repo.
 func (r *Repos) UpdateAll() error {
 	for _, table := range r.Tables {
-		log.Info("Updating %s", table.Name)
+		log.Info("Checking repo %s", table.Name)
 		rpath := filepath.Join(r.Dir, table.Name)
 		g, err := ensureRepo(table.Repo, rpath)
 		if err != nil {
 			return err
 		}
+
+		initialVersion, _ := g.Version()
 		if err := g.Update(); err != nil {
 			return err
 		}
+		diff, err := repoChartDiff(rpath, initialVersion)
+		if err != nil {
+			return err
+		}
+		printSummary(diff)
 	}
 	return nil
 }
 
+func repoChartDiff(rpath, initialVersion string) (string, error) {
+	// build git diff-tree command
+	cmd := exec.Command("git", "-C", rpath, "diff-tree", "--name-status", fmt.Sprintf("%s..HEAD", initialVersion))
+
+	log.Debug("git diff cmd: %s", cmd.Args)
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+
+	// cleanup any trailing whitespace
+	return strings.TrimSpace(string(out)), nil
+}
+
+type repoSummary map[string][]string
```

Comment:
There's no real convention for where you put non-public types, other than "not at the top of the file". The sort of basic idea is that exported (public) stuff floats toward the top, while non-exported stuff is lower down in the file. But if you look at the Go core code, you'll not see this practice followed particularly well.